CI: use relative files in coverage report

superlopuh opened the pull request:
I'm hoping it will help unconfuse codecov. I included a couple of small changes here, which is to lint the codecov.yml, and exclude lines with `assert_never` in them from coverage.

compor reviewed (approved):
How are you testing this? Are you using the `codecovcli`?

superlopuh commented:
I was mostly just iterating with the CI, now I got the cli to work, I'll iterate a bit more. I feel like I'm getting close, and this is the kind of head bashing against the wall that I'm capable of rn. I'll mark as draft for now.

superlopuh commented:
https://app.codecov.io/gh/xdslproject/xdsl/pull/4046/tree

compor commented:
Let's get it in fast, lol.

Can you pinpoint what it was? `codecov_yml_path: codecov.yml` looks suspiciously relevant

superlopuh commented:
I'm not actually sure that it was relevant, my understanding is that it was the combination of removing of "fixes" and "source", what happened was that with source we removed the parent directory, and with fixes we added it back in, but that was broken for the "docs" folder, you could see it because it went xdsl/Toy/toy.

superlopuh commented:
There may have been other things but at this point I'd rather merge and move on :)
